File: lib/src/test/java/net/mm2d/upnp/ArgumentTest.java
/*
 * Copyright (c) 2016 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

@RunWith(JUnit4.class)
public class ArgumentTest {
    @Test(expected = IllegalStateException.class)
    public void build_Actionを設定していないとException() {
        new ArgumentImpl.Builder()
                .build();
    }

    @Test(expected = IllegalStateException.class)
    public void build_Nameを設定していないとException() {
        new ArgumentImpl.Builder()
                .setAction(mock(Action.class))
                .build();
    }

    @Test(expected = IllegalStateException.class)
    public void build_RelatedStateVariableを設定していないとException() {
        new ArgumentImpl.Builder()
                .setAction(mock(Action.class))
                .setName("")
                .build();
    }

    @Test
    public void getRelatedStateVariableName_setした値が返る() {
        final String name = "name";
        final ArgumentImpl.Builder builder = new ArgumentImpl.Builder()
                .setRelatedStateVariableName(name);
        assertThat(builder.getRelatedStateVariableName(), is(name));
    }

    @Test
    public void build_Builderで指定した値が得られる() {
        final Action action = mock(Action.class);
        final String name = "name";
        final StateVariable stateVariable = mock(StateVariable.class);
        final Argument argument = new ArgumentImpl.Builder()
                .setAction(action)
                .setName(name)
                .setDirection("in")
                .setRelatedStateVariable(stateVariable)
                .build();
        assertThat(argument.getAction(), is(action));
        assertThat(argument.getRelatedStateVariable(), is(stateVariable));
        assertThat(argument.getName(), is(name));
        assertThat(argument.isInputDirection(), is(true));
    }

    @Test
    public void isInputDirection_Builderでoutを指定した場合false() {
        final Action action = mock(Action.class);
        final String name = "name";
        final StateVariable stateVariable = mock(StateVariable.class);
        final Argument argument = new ArgumentImpl.Builder()
                .setAction(action)
                .setName(name)
                .setDirection("out")
                .setRelatedStateVariable(stateVariable)
                .build();
        assertThat(argument.isInputDirection(), is(false));
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/internal/impl/ArgumentTest.kt
/*
 * Copyright (c) 2019 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.internal.impl

import com.google.common.truth.Truth.assertThat
import io.mockk.mockk
import net.mm2d.upnp.StateVariable
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@Suppress("NonAsciiCharacters", "TestFunctionName")
@RunWith(JUnit4::class)
class ArgumentTest {
    @Test(expected = IllegalStateException::class)
    fun build_Nameを設定していないとException() {
        ArgumentImpl.Builder()
                .build()
    }

    @Test(expected = IllegalStateException::class)
    fun build_RelatedStateVariableを設定していないとException() {
        ArgumentImpl.Builder()
                .setName("")
                .build()
    }

    @Test
    fun getRelatedStateVariableName_setした値が返る() {
        val name = "name"
        val builder = ArgumentImpl.Builder()
                .setRelatedStateVariableName(name)
        assertThat(builder.getRelatedStateVariableName()).isEqualTo(name)
    }

    @Test
    fun build_Builderで指定した値が得られる() {
        val name = "name"
        val stateVariable: StateVariable = mockk(relaxed = true)
        val argument = ArgumentImpl.Builder()
                .setName(name)
                .setDirection("in")
                .setRelatedStateVariable(stateVariable)
                .build()
        assertThat(argument.relatedStateVariable).isEqualTo(stateVariable)
        assertThat(argument.name).isEqualTo(name)
        assertThat(argument.isInputDirection).isTrue()
    }

    @Test
    fun isInputDirection_Builderでoutを指定した場合false() {
        val name = "name"
        val stateVariable: StateVariable = mockk(relaxed = true)
        val argument = ArgumentImpl.Builder()
                .setName(name)
                .setDirection("out")
                .setRelatedStateVariable(stateVariable)
                .build()
        assertThat(argument.isInputDirection).isFalse()
    }
}
